cumulusci/tasks/bulkdata/delete.py

```diff
@@ -38,77 +43,57 @@ def _init_options(self, kwargs):
                 "Criteria cannot be specified if more than one object is specified."
             )
         self.options["hardDelete"] = process_bool_arg(self.options.get("hardDelete"))
+        self.options["ignore_row_errors"] = process_bool_arg(
+            self.options.get("ignore_row_errors")
+        )
 
     def _run_task(self):
         for obj in self.options["objects"]:
-            self.logger.info(f"Deleting {self._object_description(obj)} ")
-            delete_job = self._create_job(obj, self.options["where"])
-            if delete_job is not None:
-                self._wait_for_job(delete_job, error_behaviour="raise")
-
-    def _create_job(self, obj, where=None):
-        # Query for rows to delete
-        delete_rows = self._query_salesforce_for_records_to_delete(obj, where)
-        if not delete_rows:
-            self.logger.info(f"  No {obj} objects found, skipping delete")
-            return
-
-        # Upload all the batches
-        operation = "hardDelete" if self.options["hardDelete"] else "delete"
-        delete_job = self.bulk.create_job(obj, operation)
-        self.logger.info(f"  Deleting {len(delete_rows)} {obj} records")
-        batch_num = 1
-        for batch in self._upload_batches(delete_job, delete_rows):
-            self.logger.info(f"    Uploaded batch {batch}")
-            batch_num += 1
-        self.bulk.close_job(delete_job)
-        return delete_job
-
-    def compose_query(self, obj, where):
-        query = f"SELECT Id FROM {obj}"
-        if where:
-            query += f" WHERE {where}"
+            query = f"SELECT Id FROM {obj}"
+            if self.options["where"]:
+                query += f" WHERE {self.options['where']}"
+
+            self.logger.info(f"Querying for {obj} objects")
+            qs = BulkApiQueryOperation(obj, {}, self, query)
+            qs.query()
+            if qs.job_result.status is not DataOperationStatus.SUCCESS:
+                raise BulkDataException(
```

Comment:
@prescod Particularly want your thoughts on these ~20 lines - I altered Delete Data to make it throw an exception if the query fails, do nothing if the query returns no results, and throw an exception if the delete job fails or if it has a row-level error and `ignore_row_errors` is unset.